### ISSUE TITLE
Uploading file fails

### ISSUE THREAD

**Issue by GochoMugo** [jdalrymple/gitbeaker#794]
**Description**

With the code below:

```ts
const content = fs.createReadStream(filepath);
await this.gitlab.Projects.upload(projectId, content, {
    metadata: {
         filename,
    },
});
```

the following error is thrown:

```
TypeError: source.on is not a function
    at Function.DelayedStream.create (<redacted>/node_modules/delayed-stream/lib/delayed_stream.js:33:10)
    at FormData.CombinedStream.append (<redacted>/node_modules/combined-stream/lib/combined_stream.js:45:37)
    at FormData.append (<redacted>/node_modules/form-data/lib/form_data.js:73:3)
    at <redacted>/node_modules/@gitbeaker/core/dist/index.js:1:4008
    at Array.forEach (<anonymous>)
    at Object.<anonymous> (<redacted>/node_modules/@gitbeaker/core/dist/index.js:1:3919)
    at <redacted>/node_modules/@gitbeaker/core/dist/index.js:1:3123
    at Object.next (<redacted>/node_modules/@gitbeaker/core/dist/index.js:1:3228)
    at <redacted>/node_modules/@gitbeaker/core/dist/index.js:1:2165
    at new Promise (<anonymous>)
```

**Possible fixes**

Usage of dependency `form-data` is wrong when handling
files i.e. on line [packages/gitbeaker-core/src/infrastructure/Utils.ts#L29](https://github.com/jdalrymple/gitbeaker/blob/8dbc11590b/packages/gitbeaker-core/src/infrastructure/Utils.ts#L29).

A possible fix:

```ts
else if ('content' in (v as any) && 'metadata' in (v as any)) form.append(k, (v as any).content, (v as any).metadata);
```

**Comment by jdalrymple** [comment]
I messed up the usage, ill fix this up before the end of the week!

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`19.6.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/19.6.0) :rocket:
